packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts

```diff
@@ -105,11 +105,12 @@ async function main() {
     claimsObject.totalRewardsDistributed
   );
 
-  console.log("\n5. Checking allowance in payment token 💸");
-  const rewardToken = new web3.eth.Contract(getAbi("ExpandedERC20"), claimsObject.rewardToken);
-  const currentRewardAllowance = await rewardToken.methods.allowance(account, options.merkleDistributorAddress).call();
-
   if (options.sendTransaction == "true") {
+    console.log("\n5. Checking allowance in payment token 💸");
+    const rewardToken = new web3.eth.Contract(getAbi("ExpandedERC20"), claimsObject.rewardToken);
+    const currentRewardAllowance = await rewardToken.methods
+      .allowance(account, options.merkleDistributorAddress)
+      .call();
     if (toBN(currentRewardAllowance).lt(toBN(MAX_UINT_VAL).div(toBN("2")))) {
```

Comment:
Nit: should use the constant that accommodates the UNI token or other tokens with non uint32 internal balance types